docs/tutorials/excel-tutorial.md

```diff
@@ -0,0 +1,956 @@
+
+3. Run the command `npm start` to start a web server running on localhost.
+
+4. Reload the task pane by closing it, and then on the **Home** menu, select **Show Taskpane** to reopen the add-in.
+
+5. If for any reason the table is not in the open worksheet, in the task pane, choose **Create Table**.
+
+6. Choose the **Filter Table** and **Sort Table** buttons, in either order.
+
+    ![Excel tutorial - Filter and Sort Table](../images/excel-tutorial-filter-and-sort-table.png)
+
+## Create a chart
+
+In this step of the tutorial, you'll create a chart using data from the table that you created previously, and then format the chart.
+
+### Chart a chart using table data
+
+1. Open the project in your code editor.
+
+2. Open the file index.html.
+
+3. Below the `div` that contains the `sort-table` button, add the following markup:
+
+    ```html
+    <div class="padding">
+        <button class="ms-Button" id="create-chart">Create Chart</button>
+    </div>
+    ```
+
+4. Open the app.js file.
+
+5. Below the line that assigns a click handler to the `sort-chart` button, add the following code:
+
+    ```js
+    $('#create-chart').click(createChart);
+    ```
+
+6. Below the `sortTable` function add the following function.
+
+    ```js
+    function createChart() {
+        Excel.run(function (context) {
+
+            // TODO1: Queue commands to get the range of data to be charted.
+
+            // TODO2: Queue command to create the chart and define its type.
+
+            // TODO3: Queue commands to position and format the chart.
+
+            return context.sync();
+        })
+        .catch(function (error) {
+            console.log("Error: " + error);
+            if (error instanceof OfficeExtension.Error) {
+                console.log("Debug info: " + JSON.stringify(error.debugInfo));
+            }
+        });
+    }
+    ```
+
+7. Replace `TODO1` with the following code. Note that in order to exclude the header row, the code uses the `Table.getDataBodyRange` method to get the range of data you want to chart instead of the `getRange` method.
+
+    ```js
+    const currentWorksheet = context.workbook.worksheets.getActiveWorksheet();
+    const expensesTable = currentWorksheet.tables.getItem('ExpensesTable');
+    const dataRange = expensesTable.getDataBodyRange();
+    ```
+
+8. Replace `TODO2` with the following code. Note the following parameters:
+
+   - The first parameter to the `add` method specifies the type of chart. There are several dozen types.
+
+   - The second parameter specifies the range of data to include in the chart.
+
+   - The third parameter determines whether a series of data points from the table should be charted row-wise or column-wise. The option `auto` tells Excel to decide the best method.
+
+    ```js
+    let chart = currentWorksheet.charts.add('ColumnClustered', dataRange, 'auto');
```

Comment:
These string literals should be switched to the `ChartType` and `ChartSeriesBy` enum values.

Reply:
Agreed; we'll address this when the content of this tutorial is re-evaluated and refactored in Q1.